aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -512,20 +512,20 @@ module aptos_framework::stake {
         let amount = coin::value<AptosCoin>(&coins);
         assert!(amount > 0, error::invalid_argument(EINVALID_STAKE_AMOUNT));
 
+        let stake_pool = borrow_global_mut<StakePool>(pool_address);
+        let (_minimum_stake, maximum_stake) = staking_config::get_required_stake(&staking_config::get());
+        let total_stake = coin::value<AptosCoin>(&stake_pool.active) +
+                          coin::value<AptosCoin>(&stake_pool.pending_active);
+        assert!(total_stake <= maximum_stake, error::invalid_argument(ESTAKE_EXCEEDS_MAX));
```

Comment:
this doesn't look right, this is only the existing stake, not the stake + amount. so people can add arbitrary amount to existing stake pool.

what's the rationale of moving this above? fail fast?

Reply:
Yeah good call. I also noticed a minor issue with this validation - it does not take into account pending_inactive. Now as pending_inactive is only moved to inactive at the end of the lockup period, a validator can exceed the max for the remaining time. I fixed this issue and added a lot of tests to make sure the max stake amount is respected in all cases.

Reply:
I also just realize that if the node join the validator set, (in pending_active), and add more stake, the coins will go into active but not reflect in the voting power

Reply:
I'm not following? on_new_epoch recalculates voting power for every validator that goes into the active set. The voting power is calculated as active + pending_inactive so this should already be taken care of for everyone, including pending_active validators?

Reply:
oh right it's regenerated, I was confused by `append(&mut validator_set.active_validators, &mut validator_set.pending_active);` and thought it uses the validator info generated when join